Release/9.0.0 (#1409)
RevenueCat/purchases-flutter: CHANGELOG-LATEST.md

* Increased min Flutter SDK to 3.22.0 and Dart min SDK to 3.4.0

* Remove freezed in favor of manually parsed models (#1368) via Toni Rico (@tonidero)
   * This removed some APIs like `toJson`, `StoreTransaction.revenueCatIdentifier`
   * This also modified our APIs to use WebPurchaseRedemptionResult. Now you will need to do:
```
WebPurchaseRedemptionResult result;
switch (result) {
  case WebPurchaseRedemptionSuccess(:final customerInfo):
    // Handle successful redemption
    break;
  case WebPurchaseRedemptionError(:final error):
    // Handle error in redemption
    break;
  case WebPurchaseRedemptionPurchaseBelongsToOtherUser():
    // Handle case where the purchase belongs to another user
    break;
  case WebPurchaseRedemptionInvalidToken():
    // Handle case where the token is invalid
    break;
  case WebPurchaseRedemptionExpired(:final obfuscatedEmail):
    // Handle case where the redemption link has expired
    break;
};
```

* Change result of purchase methods to PurchaseResult (#1408) via Toni Rico (@tonidero)

>    * Now, the result of the purchase methods is a `PurchaseResult` object, which contains the `Purchase` object instead of a `CustomerInfo` object. The `PurchaseResult` object also contains a `CustomerInfo` object, which can be used to get the latest customer information after the purchase is made aside from other useful data, including the created `StoreTransaction`.

Reviewer: I'm not sure this sentence is clear. 
> which contains the `Purchase` object

This is the part that confuses me. Did you mean the `StoreTransaction`?. In any case, maybe it could be changed to something on these lines:
```suggestion
   * The result of the purchase methods is now a `PurchaseResult` object instead of a `CustomerInfo` object. The `PurchaseResult` object contains a `CustomerInfo` object updated with the latest customer information after the purchase is made, and the `StoreTransaction` object created by the purchase.
```

Author: I messed that copy 😓. Your copy makes much more sense! Thanks!
